cpp/src/groupby/hash/compute_aggregations.cuh

```diff
@@ -67,13 +67,15 @@ rmm::device_uvector<cudf::size_type> compute_aggregations(
   auto const grid_size =
     max_occupancy_grid_size<typename SetType::ref_type<cuco::insert_and_find_tag>>(num_rows);
   auto const available_shmem_size = get_available_shared_memory_size(grid_size);
-  auto const has_sufficient_shmem =
-    available_shmem_size > (compute_shmem_offsets_size(flattened_values.num_columns()) * 2);
-  auto const has_dictionary_request = std::any_of(
-    requests.begin(), requests.end(), [](cudf::groupby::aggregation_request const& request) {
-      return cudf::is_dictionary(request.values.type());
+  auto const offsets_buffer_size  = compute_shmem_offsets_size(flattened_values.num_columns()) * 2;
+  auto const data_buffer_size     = available_shmem_size - offsets_buffer_size;
+  auto const is_shared_memory_compatible = std::all_of(
+    requests.begin(), requests.end(), [&](cudf::groupby::aggregation_request const& request) {
+      auto const size = cudf::type_dispatcher<cudf::dispatch_storage_type>(request.values.type(),
+                                                                           size_of_functor{});
+      return !cudf::is_dictionary(request.values.type()) and
+             static_cast<size_type>(data_buffer_size) >= (size * GROUPBY_CARDINALITY_THRESHOLD);
```

Comment:
Slight reorganization and added a comment. Please clang-format this after committing.
```suggestion
      if (cudf::is_dictionary(request.values.type())) {
        return false;
      }
      // Ensure there is enough buffer space to store local aggregations up to the max cardinality for shared memory aggregations
      auto const size = cudf::type_dispatcher<cudf::dispatch_storage_type>(request.values.type(),
                                                                           size_of_functor{});
      return static_cast<size_type>(data_buffer_size) >= (size * GROUPBY_CARDINALITY_THRESHOLD);
```